goto-instrument: Remove inline asm before doing various operations

tautschnig opened the pull request:
If mutations such as expanding function pointers have been performed, then
removing inline asm is reasonable as well.

smowton commented on src/goto-programs/remove_asm.cpp:
```diff
@@ -281,20 +283,29 @@ void remove_asmt::process_instruction(
+
+  if(did_something)
+  {
+    remove_skip(goto_function.body);
```
I think this would be unusual -- from what I've seen elsewhere, passes usually leave skips intact and then the driver program is expected to do one big `remove_skip` at a time of its choosing. This is especially useful in that it means the driver program can assume most passes do not alter location numbers.

tautschnig replied:
Doing `git grep remove_skip` we seem to have very much a mixed picture. I have opened #2067 to track this.

smowton replied:
At least this suggests there's no reason to change `remove_asm`'s behaviour here?

tautschnig replied:
I will move that part of the commit to a separate PR, yes.

tautschnig committed:
goto-instrument: Remove inline asm before doing various operations

If mutations such as expanding function pointers have been performed, then
removing inline asm is reasonable as well.

peterschrammel reviewed (approved):
LGTM
